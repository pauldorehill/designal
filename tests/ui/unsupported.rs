use designal::Designal;
use futures_signals::signal::Mutable;
use futures_signals::signal_vec::MutableVec;

#[derive(Designal)]
#[designal(trim_start = "Human")]
union HumanBean1 {
    taste: u8,
}

#[derive(Designal)]
#[designal(trim_start = "Human")]
struct HumanBean3 {
    #[designal(hashmap)]
    taste: Mutable<String>,
}

#[derive(Designal)]
#[designal(trim_start = "Human")]
struct HumanBean4 {
    #[designal(hashmap)]
    taste: MutableVec<String>,
}

fn main() {}
