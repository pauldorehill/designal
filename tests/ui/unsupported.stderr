error: Unions are not yet supported
 --> $DIR/unsupported.rs:7:7
  |
7 | union HumanBean1 {
  |       ^^^^^^^^^^

error: Use of `hashmap` on a non `MutableBTreeMap<K, V>`
  --> $DIR/unsupported.rs:14:16
   |
14 |     #[designal(hashmap)]
   |                ^^^^^^^

error: Use of `hashmap` on a non `MutableBTreeMap<K, V>`
  --> $DIR/unsupported.rs:21:16
   |
21 |     #[designal(hashmap)]
   |                ^^^^^^^
